File: api/src/main/java/dev/vrba/dubs/repository/ScoreRepository.java
package dev.vrba.dubs.repository;

import dev.vrba.dubs.domain.Score;
import io.micronaut.data.jdbc.annotation.JdbcRepository;
import io.micronaut.data.model.query.builder.sql.Dialect;
import io.micronaut.data.repository.CrudRepository;

@JdbcRepository(dialect = Dialect.POSTGRES)
public interface ScoreRepository extends CrudRepository<Score, Integer> {
}

File: src/main/kotlin/dev/vrba/dubsbot/repositories/ScoreRepository.kt
package dev.vrba.dubsbot.repositories

import dev.vrba.dubsbot.entities.Score
import org.springframework.data.domain.Sort
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository
import java.util.UUID

@Repository
interface ScoreRepository : CrudRepository<Score, UUID> {
    fun findByUserIdAndGuildId(userId: Long, guildId: Long): Score?
    fun findByGuildId(guildId: Long, sort: Sort): List<Score>

}